src/utils.ts

```diff
@@ -475,13 +475,22 @@ export function appendIdfAndToolsToPath() {
     }
 
     const idfPathDir = idfConf.readParameter("idf.espIdfPath");
-    modifiedEnv.IDF_PATH = idfPathDir || process.env.IDF_PATH;
+    modifiedEnv.IDF_PATH = `"${idfPathDir}"` || `"${process.env.IDF_PATH}"`;
```

Comment:
The only case you need to quote IDF_PATH is if it appears as an argument of the set/export command (`export IDF_PATH="..."`). If you are directly setting its value in the environment variable mapping, then quotes re not needed.

Reply:
By the way, for now in the CMake build system, we don't support spaces in IDF_PATH. So you might not need quoting here at all.

Reply:
Understood. For some reason we still need to do set/export in terminal since vscode API doesn't override existing IDF_PATH. 

Quoting on set/export command to keep it compatible.